### PR TITLE
[SW-2050][FollowUp] Fix Timezone Handling via Conversions to UTC (#1950)

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: core/src/main/scala/ai/h2o/sparkling/backend/converters/TimeZoneConversions.scala [hunk 1]
```diff
@@ -25,15 +25,15 @@ import org.apache.spark.sql.catalyst.util.DateTimeUtils
 trait TimeZoneConversions {
   def sparkTimeZone: TimeZone
 
-  def fromSparkTimeZoneToUTC(timestamp: Long): Long = DateTimeUtils.fromUTCTime(timestamp, sparkTimeZone.getID)
+  def fromSparkTimeZoneToUTC(timestamp: Long): Long = DateTimeUtils.toUTCTime(timestamp, sparkTimeZone.getID)
```
Comment (jakubhava) [inline core/src/main/scala/ai/h2o/sparkling/backend/converters/TimeZoneConversions.scala]: I believe we were using opposite methods in this class

File: core/src/test/scala/org/apache/spark/h2o/converters/DataFrameConverterTest.scala [hunk 2]
```diff
@@ -102,20 +103,33 @@ class DataFrameConverterTest extends FunSuite with SharedH2OTestContext {
...
+    assert(df.collect().sameElements(rddBack.collect()))
+    h2oFrame.delete()
+  }
+
+  test("H2OFrame[T_TIME] UTC to DataFrame[TimestampType]") {
     val fname: String = "testTime.hex"
     val colNames: Array[String] = Array("C0")
     val chunkLayout: Array[Long] = Array(2L, 2L)
     val data: Array[Array[Long]] = Array(Array(1428517563L, 1428517564L), Array(1428517565L, 1428517566L))
+      .map(_.map(t => DateTimeUtils.toUTCTime(t * 1000, SparkTimeZone.current().getID)))
```
Comment (jakubhava) [inline core/src/test/scala/org/apache/spark/h2o/converters/DataFrameConverterTest.scala]: We need to simulate H2O import ( treat time as in UTC)

File: core/src/main/scala/ai/h2o/sparkling/backend/converters/TimeZoneConversions.scala (hunk 1 above)
Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/backend/converters/TimeZoneConversions.scala]: This was made like this on purpose. It seems that Spark internal methods has opposite meaning.

Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/backend/converters/TimeZoneConversions.scala]: The python tests I added should show us the correct behavior

**Pull request merged**